nexusutils/dataconverter/writer.py

```diff
@@ -202,22 +207,27 @@ def __nxdl_to_attrs(self, path: str = '/') -> dict:
 
         # Fetch values for required attributes requested by the NXDL
         for attr_name in elem.findall(f"{self.nxs_namespace}attribute"):
-            elem.attrib[attr_name.get('name')] = self.data[f"{path}/@{attr_name.get('name')}"] or ''
+            key = f"{path}/@{attr_name.get('name')}"
+            if key in self.data:
+                elem.attrib[attr_name.get('name')] = self.data[key]
+            else:
+                elem.attrib[attr_name.get('name')] = ''
```

Comment:
This can be removed as it isn't used and it doesn't make sense to add empty attributes.

Reply:
Done